qiskit_nature/mappers/second_quantization/bksf.py

```diff
@@ -0,0 +1,532 @@
+       If `expand_number_op` is `True`, then factors of `N` are expanded.
+
+    Raises:
+       ValueError: if any character in `term_str` is not one of "+-IN".
+    """
+    (n_number, n_raise, n_lower) = (0, 0, 0)
+    facs = []
+    for i, c in enumerate(term_str):
+        if c == "I":
+            continue
+        if c == "+":
+            n_raise += 1
+            facs.append((i, "+"))
+        elif c == "-":
+            n_lower += 1
+            facs.append((i, "-"))
+        elif c == "N":
+            n_number += 1
+            if expand_number_op:
+                facs.append((i, "+"))
+                facs.append((i, "-"))
+            else:
+                facs.append((i, "N"))
+        else:
+            raise ValueError("Unexpected operator ", c, " in term.")
+
+    return (n_number, n_raise, n_lower), facs
+
+
+def _interaction_type(n_number: int, n_raise: int, n_lower: int) -> TermType:
+    """
+    Return a `TermType` instance describing the type of interaction given the number of
+    number, raising, and lowering operators.
+
+    Args:
+       `n_number`: the number of number operators
+       `n_raise`: the number of raising operators
+       `n_lower`: the number of lowering operators
+
+    Returns:
+      TermType: The type of interaction.
+
+    Raises:
+      ValueError: if the numbers of operators don't describe a one- or two-body term from
+      an electronic Hamiltonian.
+    """
+    if n_raise == 0 and n_lower == 0:
+        if n_number == 1:
+            return TermType.NUMBER
+        elif n_number == 2:
+            return TermType.COULOMB_EXCHANGE
+        else:
+            raise ValueError("unexpected number of number operators: ", n_number)
+    elif n_raise == 1 and n_lower == 1:
+        if n_number == 1:
+            return TermType.NUMBER_EXCITATION
+        elif n_number == 0:
+            return TermType.EXCITATION
+        else:
+            raise ValueError("unexpected number of number operators: ", n_number)
+    elif n_raise == 2 and n_lower == 2:
+        return TermType.DOUBLE_EXCITATION
+    else:
+        raise ValueError("unexpected number of operators: ", n_number)
+
+
+def _get_adjacency_matrix(fer_op: FermionicOp) -> np.ndarray:
+    """
+    Return an adjacency matrix specifying the edges in the BKSF graph for the
+    operator `fer_op`.
+
+    The graph is undirected, so we choose to return the edges in the upper triangle.
+    (There are no self edges.). The lower triangle are all `False`.
+
+    Returns:
+          numpy.ndarray(dtype=bool): edge_matrix the adjacency matrix.
+    """
+    n_modes = fer_op.register_length
+    edge_matrix = np.zeros((n_modes, n_modes), dtype=bool)
+    for term in fer_op.to_list():
+        _add_edges_for_term(edge_matrix, _operator_string(term))
+    return edge_matrix
+
+
+def _add_one_edge(edge_matrix: np.ndarray, i: int, j: int) -> None:
+    """
+    Add an edge from lesser index to greater. This maintains the upper triangular structure.
```

Comment:
Missing `Args` and `Raises` documentation

Reply:
The same comment of @mrossinek applies for many private methods. I am not sure if it is needed for private methods but would be good to include it.

Reply:
> applies for many private methods
Yes. Since you made these comments, I have read the style guide on docstrings, and revised all of the docstrings. However, this one in particular does not yet conform.